test/rubygems/test_gem.rb

```diff
@@ -29,6 +29,17 @@ def setup
     util_remove_interrupt_command
   end
 
+  def teardown
+    super
+
+    change_version @version if defined? @version
```

Comment:
using `||=` could work, as I suggested below, else resetting to `Gem::Version.new(Gem::VERSION)`